### PR TITLE
Added cstor-istgt-image target into cstor makefile and updated gitign…

### PR DESCRIPTION
**Pull request by Moteesh**
…ore.

Signed-off-by: moteesh <moteesh@gmail.com>

This commit adds the cstor-istgt-image target into cstor makefile so that when make is invoked
istgt docker image will be built along with other images.

Changes to be committed:
A       .gitignore
A       Dockerfile.Istgtimage
M       GNUmakefile
A       cstor-istgt
A       entrypoint-istgtimage.sh
A       push-istgtimage

1.Why is this change necessary ?
To build istgt docker image and push it to dockerhub upon making cstor.
2. How does this change address the issue ?
This commit adds the docker file and required scripts to build istgt and push image to dockerhub. GNUMakefile has been added with a new target called cstor-istgt-image which builds the docker image and push the image to dockerhub
3. How to verify this change ?
Do make in cstor directory and verify the docker image of istgt gets built.
4. What side effects does this change have ?
5. Other details

### CODE REVIEW

File: Dockerfile.Istgtimage [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#
+# This Dockerfile builds cstor istgt container running istgt from istgt base image
+#
+
+FROM openebs/istgt_base:latest
```
Comment (gkGaneshR) [inline Dockerfile.Istgtimage]: Can you pls tell where this istgt_base image is located?

**Comment by vishnuitta** [comment]
@payes please review these changes to make docker image for istgt binary

File: Dockerfile.Istgtimage [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ARG BUILD_DATE
+LABEL org.label-schema.name="cstor"
+LABEL org.label-schema.description="OpenEBS cstor"
+LABEL org.label-schema.url="http://www.openebs.io/"
+LABEL org.label-schema.vcs-url="https://github.com/openebs/cstor"
+LABEL org.label-schema.schema-version="1.0"
+LABEL org.label-schema.build-date=$BUILD_DATE
+
+ENTRYPOINT entrypoint-istgtimage.sh
+EXPOSE 3260
```
Comment (vishnuitta) [inline Dockerfile.Istgtimage]: @payes do we need any other ports to open?

**Comment by Moteesh** [comment]
I added .gitignore file to ignore the checked out code of istgt from revision control while commiting.

File: entrypoint-istgtimage.sh [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+trap 'call_exit $LINE_NO' EXIT
+
+call_exit()
+{
+echo "at call_exit.."     
+echo  "exit code:" $?
+echo "reference: "  $0 
+}
+
+exec /usr/local/bin/init.sh volname=vol1 portal=127.0.0.1 path=/tmp/cstor size=10g externalIP=127.0.0.1 replication_factor=3 consistency_factor=2 &
```
Comment (vishnuitta) [inline entrypoint-istgtimage.sh]: why are we having running init.sh?

Comment (payes) [reply entrypoint-istgtimage.sh]: init.sh is the script which will update istgt.conf file with the required parameters and kick start the istgt process inside the container.

File: entrypoint-istgtimage.sh [hunk 4]
```diff
@@ -10,7 +10,7 @@ echo  "exit code:" $?
 echo "reference: "  $0 
 }
 
-exec /usr/local/bin/init.sh volname=vol1 portal=127.0.0.1 path=/tmp/cstor size=10g externalIP=127.0.0.1 replication_factor=3 consistency_factor=2 &
+exec /usr/local/bin/istgt &
```
Comment (payes) [inline entrypoint-istgtimage.sh]: lets also start rsyslog over here

**Comment by vishnuitta** [comment]
Thanks guys for reviewing. Thanks Moteesh for the changes.

**Pull request merged**